test/runtime/runtime.shared.js

```diff
@@ -0,0 +1,271 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { clear_loops, flush, set_now, set_raf } from 'svelte/internal';
+import { setImmediate } from 'timers/promises';
+import glob from 'tiny-glob/sync.js';
+import { afterAll, assert, beforeAll, describe, it } from 'vitest';
+import { compile } from 'svelte/compiler';
+import { create_loader, mkdirp, try_load_config } from '../helpers.js';
+import { setup_html_equal } from '../html_equal.js';
+
+let unhandled_rejection = false;
+function unhandledRejection_handler(err) {
+	unhandled_rejection = err;
+}
+
+const listeners = process.rawListeners('unhandledRejection');
+
+const { assert_html_equal, assert_html_equal_with_options } = setup_html_equal({
+	removeDataSvelte: true
+});
+
+beforeAll(() => {
+	process.prependListener('unhandledRejection', unhandledRejection_handler);
+});
+
+afterAll(() => {
+	process.removeListener('unhandledRejection', unhandledRejection_handler);
+});
+
+const failed = new Set();
+
+async function run_test(dir) {
+	if (dir[0] === '.') return;
+
+	const config = await try_load_config(`${__dirname}/samples/${dir}/_config.js`);
+	const solo = config.solo || /\.solo/.test(dir);
+
+	const it_fn = config.skip ? it.skip : solo ? it.only : it;
+
+	it_fn.each`
+			hydrate  | from_ssr_html
+			${false} | ${false}
+			${true}  | ${false}
+			${true}  | ${true}
+		`(`${dir} hydrate: $hydrate, from_ssr: $from_ssr_html`, async ({ hydrate, from_ssr_html }) => {
+		if (hydrate && config.skip_if_hydrate) return;
+		if (hydrate && from_ssr_html && config.skip_if_hydrate_from_ssr) return;
+
+		if (failed.has(dir)) {
+			// this makes debugging easier, by only printing compiled output once
+			assert.fail(`skipping ${dir}, already failed`);
+		}
+
+		unhandled_rejection = null;
+
+		const cwd = path.resolve(`${__dirname}/samples/${dir}`);
+
+		const compileOptions = Object.assign({}, config.compileOptions || {}, {
+			format: 'cjs',
+			hydratable: hydrate,
+			immutable: config.immutable,
+			accessors: 'accessors' in config ? config.accessors : true
+		});
+
+		const load = create_loader(compileOptions, cwd);
+
+		let mod;
+		let SvelteComponent;
+
+		let unintendedError = null;
+
+		if (config.expect_unhandled_rejections) {
+			listeners.forEach((listener) => {
+				// @ts-expect-error
+				process.removeListener('unhandledRejection', listener);
+			});
+		}
+
+		await Promise.resolve()
```

Comment:
why is the code here inside `Promise.resolve().then`? Is it to cause a new tick to happen first? could you just do something like `await 0;` instead? I'm wondering if you could then move the code the `then` which would make it easier to read on a small screen by reducing the indentation